option.go

```diff
@@ -174,3 +198,5 @@ func ScopedRateLimiter(l *rate.Limiter) SyncOption {
 		sc.rateLimiter = l
 	}
 }
+
+// TODO: add scoped variations of segmented sync options.
```

Comment:
Deferring this to a separate PR since I don't believe it's a blocker for resolving memory issues in storetheindex.

Reply:
Going to implement this since it is useful in storetheindex today.